lib/Conversion/TorchToLinalg/Reduction.cpp

```diff
@@ -228,14 +233,176 @@ static Value createLinalgPayloadCalculationForReduceOp(
       if (intType.isSigned())
         return b.create<arith::MaxSIOp>(loc, self, result);
     }
+  } else if (isa<AtenLinalgVectorNormOp>(op)) {
+    // This creates payload for only the first of the two linalg.generic ops.
+    // TODO: Short-circuit operations if `ord` is zero or one.
+    auto elem = payloadArgs[0];
+    auto result = payloadArgs[1];
+    auto self = convertScalarToDtype(b, loc, elem, resultElementType);
+    auto abs = b.create<math::AbsOp>(loc, self);
+    auto ord = convertScalarToDtype(b, loc, operands[1], resultElementType);
+    auto pow = b.create<math::PowFOp>(loc, abs, ord);
+    return b.create<arith::AddFOp>(loc, pow, result);
   }
-  op->emitError("unimplemented lowering in "
-                "createLinalgPayloadCalculationForReduceOp");
+  op->emitError("unimplemented lowering in createLinalgPayloadForReduceOp");
   return nullptr;
 }
 
 namespace {
 class ConvertReductionOp : public ConversionPattern {
+private:
+  /// Given a reduction operation that has the `keepdim` attribute and the
+  /// (optional) `dim` attribute, return the source tensor operand and the
+  /// literal values of the attributes or failure otherwise.
+  template <typename T>
+  FailureOr<torch_to_linalg::ReductionOpInfo>
+  computeReductionOpInfoForDimVariantOp(
+      T op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const {
+    auto opInfo = torch_to_linalg::ReductionOpInfo{false, Value{}, {}};
+    opInfo.tensorOperand = operands[0];
```

Comment:
Sadly, same problem here about value conversion -- we want the values after conversion, which, as I understand, can only be obtained using `operands`.

Reply:
Yeah, this can be solved using the Adaptor. I think in this case you can do

```cpp
typename T::Adaptor adaptor(operands);
opInfo.tensorOperand = adaptor.self();
```

Reply:
Oh wow, thanks!  I confess I didn't know enough about adaptors until you pointed out their usage here.  Fixed in commit 6e742e81.